client/signup/steps/theme-selection/index.jsx

```diff
@@ -33,6 +33,12 @@ module.exports = React.createClass( {
 		};
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.signupDependencies !== this.props.signupDependencies && nextProps.signupDependencies.theme === this.props.signupDependencies.theme ) {
```

Comment:
The theme check is to avoid refreshing themes when selecting a theme (which updates `signupDependencies.theme`).